Tank/src/Components/AnimationComponent.cs

```diff
@@ -0,0 +1,159 @@
+        public bool PingPong
+        {
+            get => pingPong;
+            set => pingPong = value;
+        }
+
+        /// <summary>
+        /// Is the animation played in a loop
+        /// </summary>
+        private bool loop;
+
+        /// <summary>
+        /// Public accessor if the animation is played in a loop
+        /// </summary>
+        public bool Loop
+        {
+            get => loop;
+            set => loop = value;
+        }
+
+        /// <summary>
+        /// Current index of the frame
+        /// </summary>
+        private int currentIndex;
+
+        /// <summary>
+        /// Public accessor to the current frame index
+        /// </summary>
+        public int CurrentIndex
+        {
+            get => currentIndex;
+            set => currentIndex = value;
+        }
+
+        /// <summary>
+        /// The seconds used a threshold to move the index
+        /// </summary>
+        private float timeThreshold;
+
+        /// <summary>
+        /// Public access to the threshold for moving the index
+        /// </summary>
+        public float TimeThreshold
+        {
+            get => timeThreshold;
+            set => timeThreshold = value;
+        }
+
+        /// <summary>
+        /// The update frequency for the index in seconds
+        /// </summary>
+        private readonly float frameSeconds;
+
+        /// <summary>
+        /// Readonly access to the update frequency for the index in seconds
+        /// </summary>
+        public float FrameSeconds => frameSeconds;
+
+        /// <summary>
+        /// All the rectangles in the sprite making up this animation
+        /// </summary>
+        private readonly List<Rectangle> spriteSources;
+
+        /// <summary>
+        /// Readonly access to the sprite positions making up this animation
+        /// </summary>
+        public List<Rectangle> SpriteSources => spriteSources;
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        /// <param name="frameSeconds">The second threshold used to update the index</param>
+        /// <param name="spriteSources">The frames in the sprite</param>
+        public AnimationComponent(float frameSeconds, List<Rectangle> spriteSources)
+            : this(0, frameSeconds, spriteSources)
+        {
+
+        }
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        /// <param name="startIndex">The index to start the animation at</param>
+        /// <param name="frameSeconds">The second threshold used to update the index</param>
+        /// <param name="spriteSources">The frames in the sprite</param>
+        public AnimationComponent(int startIndex, float frameSeconds, List<Rectangle> spriteSources)
+        {
+            currentIndex = startIndex;
+            this.frameSeconds = frameSeconds;
+            this.spriteSources = spriteSources;
+            this.ForwardDirection = true;
```

Comment:
Allow multiple should be on true since a entity could contain multiple animations per sprite sheet